kore/src/Kore/Equation/Equation.hs

```diff
@@ -226,22 +228,44 @@ refreshVariables
     (from @_ @(Set (UnifiedVariable _)) -> avoid)
     equation@(Equation _ _ _ _ _)
   =
-    let rename = Fresh.refreshVariables avoid originalFreeVariables
-        mapElemVars elemVar =
-            case Map.lookup (ElemVar elemVar) rename of
-                Just (ElemVar elemVar') -> elemVar'
-                _ -> elemVar
-        mapSetVars setVar =
-            case Map.lookup (SetVar setVar) rename of
-                Just (SetVar setVar') -> setVar'
-                _ -> setVar
-        subst = TermLike.mkVar <$> rename
+    let rename =
```

Comment:
I can't help notice that the complexity of the code has grown by a non-trivial margin. From 10 lines with easy-to-read-and-write code, we got to more than 30 of code with lenses

Reply:
Yes, it's unfortunate, but temporary: the lenses are a consequence of switching between representations of variables, and after I finish the next part of the changes, then all the lenses go away.